product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/testng/listeners/TestExecutionListener.java

```diff
@@ -0,0 +1,237 @@
+
+            backendURL = infraProperties.getProperty(ScenarioConstants.CARBON_SERVER_URL) +
+                    (infraProperties.getProperty(ScenarioConstants.CARBON_SERVER_URL).endsWith("/") ? "" : "/");
+            mgtConsoleUrl = infraProperties.getProperty(ScenarioConstants.MGT_CONSOLE_URL);
+
+            setKeyStoreProperties();
+
+            // login
+            try {
+                AuthenticatorClient authenticatorClient = new AuthenticatorClient(backendURL);
+                sessionCookie = authenticatorClient.login("admin", "admin", getServerHost(mgtConsoleUrl));
+            } catch (Exception e) {
+                throw new RuntimeException("Login failed", e);
+            }
+
+            try {
+                String[] cAppList = carbonAppListStr.split(",");
+                for (String cApp : cAppList) {
+                    // deploy carbon application
+                    deployCarbonApplication(cApp.trim());
+                }
+            } catch (RemoteException | ApplicationAdminExceptionException e) {
+                throw new RuntimeException("Error occurred while deploying carbon application : " + carbonAppListStr, e);
+            }
+        }
+    }
+
+    @Override
+    public void onExecutionFinish() {
+        log.info("#onExecutionFinish");
+    }
+
+    /**
+     * Function to upload carbon application
+     *
+     * @param carFileName
+     * @return
+     * @throws RemoteException
+     */
+    private void deployCarbonApplication(String carFileName) throws RemoteException, ApplicationAdminExceptionException {
+
+        if (standaloneMode) {
+            if (applicationAdminClient == null) {
+                applicationAdminClient = new ApplicationAdminClient(backendURL, sessionCookie);
+            }
+
+            if (!checkCAppDeployed(applicationAdminClient, carFileName)) {
+
+                log.info("Deploying carbon application : " + carFileName);
+                // If standalone mode, deploy the CApp to the server
+                String cappFilePath = carbonApplicationsDir + File.separator + carFileName + ScenarioConstants.CAPP_EXTENSION;
+
+                if (carbonAppUploaderClient == null) {
+                    carbonAppUploaderClient = new CarbonAppUploaderClient(backendURL, sessionCookie);
+                }
+
+                DataHandler dh = new DataHandler(new FileDataSource(new File(cappFilePath)));
+                // Upload carbon application
+                carbonAppUploaderClient.uploadCarbonAppArtifact(carFileName + ScenarioConstants.CAPP_EXTENSION, dh);
+
+                //TODO - This thread sleep is added temporarily to wait until the ESB Instances sync in the cluster in clustered test environment
+                if (!Boolean.valueOf(infraProperties.getProperty(ScenarioConstants.STANDALONE_DEPLOYMENT))) {
+                    log.info("Waiting for artifacts synchronized across cluster nodes");
+                    try {
+                        Thread.sleep(120000);
```

Comment:
Shall we have a constant for this? We'll be able to use same value whenever we meet the same requirement if we have a constant

Reply:
this is temporary as described in TODO comment